### PR TITLE
Allow enabling temp user creation for IP Masking QA

### PR DESCRIPTION
**Pull request by kostajh**
Bug: T326651

### CODE REVIEW

**Review by Tchanders — COMMENTED** [review]
This looks good to me - thanks @kostajh

File: new/postinstall.sh [hunk 1]
```diff
@@ -98,6 +98,11 @@ if [ "${USE_PROXY}" = "1" ]; then
 	done
 fi
 
+# Enable temp user creation if selected
```
Comment (edg2s) [inline new/postinstall.sh:101]: "'Enable temporary user account creation (IP Masking)"

File: new/postinstall.sh [hunk 2]
```diff
@@ -98,6 +98,11 @@ if [ "${USE_PROXY}" = "1" ]; then
...
+if [ "${USE_TEMPUSER}" = "1" ]; then
+  cp $PATCHDEMO/localsettings/feature-tempuser.php $PATCHDEMO/wikis/$NAME/w/settings.d
```
Comment (edg2s) [inline new/postinstall.sh:103]: use tabs

**Review by edg2s — CHANGES_REQUESTED** [review]

**Comment by MatmaRex** [comment]
Thanks for doing this!

Just for the record, I'm not a fan of adding more options to the Patch demo form – it's already so long, it barely fits on a 1080p screen. I'd usually suggest creating a "do not merge" Gerrit changeset somewhere that changes the MediaWiki defaults, and can be combined with other patches for Patch demo.

I am, however, a fan of the IP Masking project, and if this helps move it forward, then I'm for it.

image

**Comment by edg2s** [comment]
I have a local patch to collapse everything below "Choose included repos" into an "Advanced" section. Thoughts @MatmaRex ?

**Comment by MatmaRex** [comment]
Sounds okay to me, but I don't know how everyone who wanted these options added will feel about it ;)

**Comment by kostajh** [comment]
> I'd usually suggest creating a "do not merge" Gerrit changeset somewhere that changes the MediaWiki defaults, and can be combined with other patches for Patch demo.

That seems tricky for non-technical users who want to preview or QA patches.

> I have a local patch to collapse everything below "Choose included repos" into an "Advanced" section.

That sounds like a good idea to me.

**Pull request merged**